src/library/scala/collection/mutable/BitSet.scala

```diff
@@ -170,6 +183,99 @@ class BitSet(protected[collection] final var elems: Array[Long])
   override def zip[B](that: IterableOnce[B])(implicit @implicitNotFound(collection.BitSet.zipOrdMsg) ev: Ordering[(Int, B)]): SortedSet[(Int, B)] =
     super.zip(that)
 
+  def oldAddAll(xs: IterableOnce[Int]): this.type = xs match {
+    case bs: collection.BitSet => this |= bs
+    case _ => super.addAll(xs)
+  }
+
+  override def addAll(xs: IterableOnce[Int]): this.type = xs match {
+    case bs: collection.BitSet => this |= bs
+    case range: Range if (range.step == 1 || range.step == -1) =>
+      if (range.nonEmpty) {
+        val start = range.min
+        if (start >= 0) {
+          val end = range.max
+          val endIdx = end >> LogWL
+          ensureCapacity(endIdx)
+
+          if (range.step == 1 || range.step == -1) {
```

Comment:
this appears to duplicate the guard in the `case` clause

Reply:
Ah yes thanks! I recently brought the guard down out of the case statement so that even if the step is not by `+- 1`, we can still take the opportunity to perform an `ensureCapacity` by performing the fast `range.max` operation, and then falling back to `super.addAll`